scio-parquet/src/main/scala/com/spotify/scio/parquet/read/ParquetRead.scala

```diff
@@ -43,8 +58,92 @@ object ParquetRead {
     readSupportFactory: ReadSupportFactory[T],
     conf: SerializableConfiguration,
     projectionFn: SerializableFunction[T, R]
-  ): ParDo.SingleOutput[ReadableFile, R] = {
+  ): PTransform[PCollection[ReadableFile], PCollection[R]] = {
     val sdf = new ParquetReadFn[T, R](readSupportFactory, conf, projectionFn)
-    ParDo.of(sdf)
+    val tfx: PTransform[PCollection[_ <: ReadableFile], PCollection[R]] = ParDo.of(sdf)
+
+    tfx.asInstanceOf[PTransform[PCollection[ReadableFile], PCollection[R]]]
+  }
+
+  def readTyped[T: ClassTag: Coder: ParquetType](
+    predicate: FilterPredicate = null,
+    conf: Configuration = null
+  ): PTransform[PCollection[ReadableFile], PCollection[T]] = readTyped[T, T](
+    identity,
+    predicate,
+    conf
+  )
+
+  def readTyped[T: ClassTag: Coder: ParquetType, R](
+    projectionFn: T => R,
+    predicate: FilterPredicate,
+    conf: Configuration
+  ): PTransform[PCollection[ReadableFile], PCollection[R]] = {
+    val configuration = ParquetConfiguration.ofNullable(conf)
+    Option(predicate).foreach(p => ParquetInputFormat.setFilterPredicate(configuration, p))
+
+    val cleanedFn = Functions.serializableFn(ClosureCleaner.clean(projectionFn))
+    readFiles(ReadSupportFactory.typed[T], new SerializableConfiguration(configuration), cleanedFn)
+  }
+
+  def readAvroGenericRecordFiles(
+    schema: Schema,
+    predicate: FilterPredicate = null,
+    conf: Configuration = null
+  ): PTransform[PCollection[ReadableFile], PCollection[GenericRecord]] =
+    readAvroGenericRecordFiles[GenericRecord](schema, identity, predicate, conf)
+
+  def readAvroGenericRecordFiles[T](
+    schema: Schema,
+    projectionFn: GenericRecord => T,
```

Comment:
This does not look to be a projection: We are reading the whole schema data even if the function selects a single field of the GenericRecord.